app/assets/javascripts/application.js

```diff
@@ -20,4 +20,10 @@
 //= require nprogress
 //= require nprogress-turbolinks
 //= require main
-//= require jquery.sidr.min
\ No newline at end of file
+//= require jquery.sidr.min
+
+$(document).ajaxError(function (e, xhr, settings) {
```

Comment:
well... This is kinda debatable :)
To me, it seems like one would always expect the result of an AJAX request that returns a 401 to redirect him to login page since it would be the same effect he'd get using normal HTML request.
Your opinion still makes sense though that one might want to handle it differently but I just can't think of a real use-case for that now. Let me apply your suggestion anyways :)